doc/developer-guide/api/functions/TSRecYAMLConfigParse.en.rst

```diff
@@ -0,0 +1,147 @@
+
+   Record field callback (called on every parsed record field(last node field on each YAML map))
+
+.. function:: TSReturnCode TSRecYAMLConfigParse(TSYaml node, TSYAMLRecNodeHandler handler, void *data);
+
+   Parse a YAML node following the record structure. Handler will be called on every final field.
+
+
+Description
+===========
+
+
+   ``TSYAMLRecCfgFieldData``
+
+   This contains information about the parsed record field. This will be passed back to the `TSYAMLRecNodeHandler`
+   when you call `TSRecYAMLConfigParse`. This class holds the record name as well as the YAML node in case the caller
+   wants to use this information to manipulate the record.
+
+.. var:: const char* field_name
+
+   A null-terminated string with the YAML field name. This holds the name of the scalar field. This is mostly to be used for logging.
+
+.. var:: const char* record_name
+
+    A null-terminated string with the record name which was built when parsing the YAML file. Example: `proxy.config.diags.debug.enabled`.
+
+.. var:: TSYaml value_node
+
+   A `YAML::Node` type holding the value of the field. Field value must be extracted from here. You can use the `YAML::Node` API to deduce
+   the type, etc.
+
+   :c:tye: TSYAMLRecNodeHandler
+
+   Callback function for the caller to deal with each parsed node. ``cfg`` holds the details of the parsed field. `data` can be used to
+   pass information along.
+
+   :c:func: TSRecYAMLConfigParse
+
+   Parse a YAML node following the record structure internals. On every scalar node the `handler` callback will be
+   invoked with the appropriate parsed fields. `data` can be used to pass information along to every callback, this could be
+   handy when you need to read/set data inside the `TSYAMLRecNodeHandler` to be read at a later stage.
+
+
+Example:
+
+   .. code-block:: yaml
+
+      ts:
+        plugin_x:
+          field_1: 1
+          data:
+            data_field_1: XYZ
+
+
+   `TSRecYAMLConfigParse` will parse the node and will call back on each field (`field_1` and `data_field_1`)
+
+
+   A coding example using the above yaml file would look like:
+
+
+   .. code-block:: cpp
+
+      TSReturnCode
+      field_handler(const TSYAMLRecCfgFieldData *cfg, void *data/*optional*/)
+      {
+         YAML::Node value = *reinterpret_cast<YAML::Node *>(cfg->value_node);
+
+         /*
+          Th callback function will be invoked twice:
+
+          1:
+            cfg->field_name = field_1
+            cfg->record_name = ts.plugin_x.field_1
+
+          2:
+            cfg->field_name = data_field_1
+            cfg->record_name = ts.plugin_x.data.data_field_1
+
+
+          If we need to check the type, we either get the type from the YAML::Node::Tag if set, or from @c TSHttpTxnConfigFind
+         */
+
+         return TS_SUCCESS;
+      }
+
+      void my_plugin_function_to_parse_a_records_like_yaml_file(const char* path) {
+         try {
```

Comment:
Is the `try` necessary? Shouldn't the internal API take care of that? But I suppose you'd need a mark in the callback to indicate a parse failure. Hmmm.

Reply:
this is to cover a YAML exception in this function as in this example the function is calling `YAML::LoadFile(path)`
The internal API is dealing with the errors and returing `TS_FAILURE` in that case. It will also drop a message under the `plugin` tag(as many others seems to do).

In any case I think it would be a good idea to add another safety net inside the api implementation  as we are dealing with yaml nodes and this could potentially be an issue. Thanks Alan.